plugin/trino-loki/src/main/java/io/trino/plugin/loki/LokiMatrixRecordCursor.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.loki;
+
+import io.trino.plugin.loki.model.Matrix;
+import io.trino.plugin.loki.model.MetricPoint;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkState;
+import static io.trino.spi.type.DateTimeEncoding.packDateTimeWithZone;
+
+public class LokiMatrixRecordCursor
+        extends LokiRecordCursor
+{
+    private final Iterator<LokiMatrixRecordCursor.Point> metricItr;
```

Comment:
Can we avoid using abbrevation ?

Reply:
You mean something else other than `metricItr`? Sure.